### DISCUSSION TITLE
Running multiple dagruns in a second

### DISCUSSION THREAD

**Discussion by NonlinearNimesh** [apache/airflow#27659] [Q&A]
I am triggering an airflow dag with an api but it is throwing me this response
```
{
  "detail": "DAGRun with DAG ID: 'working_1' and DAGRun logical date: '2022-11-03 06:35:14+00:00' already exists",
  "status": 409,
  "title": "Conflict",
  "type": "https://airflow.apache.org/docs/apache-airflow/2.3.3/stable-rest-api-ref.html#section/Errors/AlreadyExists"
}
```
and this is becasue i am triggering multiple dag runs in 1 seconds with same DAG ID. Below is my code
```
data = {
    "conf": {},
    "replace_microseconds":"false"
    "dag_run_id": str(uuid.uuid4()),
    "logical_date": str(formatted_date),
}
headers={
    'Content-type':'application/json',
    'Accept':'application/json'
}
json_payload = json.dumps(data)
r = requests.post("http://localhost:8080/api/v1/dags/working_1/dagRuns", auth=HTTPBasicAuth("airflow", "airflow"), data=json_payload, headers=headers)
print(r.status_code)
print(r.text)
```
i tried looking at at the airflow documentation where they have mentioned that using "replace_microseconds" and setting it to false will get the job done. But, it doesn't solved my problem. Also, becasue of some constraints i cannot put a time.sleep(1) between every dag run. Also there is a fix datetime format of logical_date so i am not able to add the microseconds into it.

If anyone can help me what sort of chage will get the job done will be great

**Comment by potiuk** [comment]
You should generate a "dag_run_id" yourself and pass it in the API:

https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run

The "replace_microseconds" was used in the old, already deprecated, experimental API, not in the stable API of ours. In Airlflow 2.2. there was a change that the "execution_date" used previously as part of uniqueness key has been replaced with "run_id" and you are free to generate run_id in the way that will allow you to maintain uniqueness of triggered dags.

You just have to make sure (and this is on your side) that every uniquely triggered run has a unique run_id.
